web/satellite/src/components/accessGrants/CreateAccessModal.vue

```diff
@@ -0,0 +1,550 @@
+import PermissionsIcon from '@/../static/images/accessGrants/create-access_permissions.svg';
+import NameIcon from '@/../static/images/accessGrants/create-access_name.svg';
+import BucketsIcon from '@/../static/images/accessGrants/create-access_buckets.svg';
+import DateIcon from '@/../static/images/accessGrants/create-access_date.svg';
+import NotesIcon from '@/../static/images/accessGrants/create-access_notes.svg';
+import Chevron from '@/../static/images/accessGrants/chevron.svg';
+
+import { ACCESS_GRANTS_ACTIONS } from '@/store/modules/accessGrants';
+import { BUCKET_ACTIONS } from "@/store/modules/buckets";
+import { APP_STATE_MUTATIONS } from "@/store/mutationConstants";
+
+import { RouteConfig } from '@/router';
+
+// @vue/component
+@Component({
+    components: {
+        VCheckbox,
+        VButton,
+        DurationSelection,
+        BucketsSelection,
+        BucketNameBullet,
+        CloseCrossIcon,
+        TypesIcon,
+        PermissionsIcon,
+        NameIcon,
+        BucketsIcon,
+        DateIcon,
+        NotesIcon,
+        Chevron,
+    },
+})
+export default class CreateAccessModal extends Vue {
+    @Prop({default: 'Default'})
+    private readonly label: string;
+    @Prop({default: 'Default'})
+    private readonly defaultType: string;
+    @Prop({default: () => { return; }})
+    private readonly onClose: () => void;
+
+    private checkedType = '';
+    private showAllPermissions = {show: false, position: "up"};
+    private permissionsList = ["read","write","list","delete"];
+    private accessName = '';
+    private checkedPermissions = {read: false, write: false, list: false, delete: false};
+    private selectedPermissions : string[] = [];
+    private allPermissionsClicked = false;
+    public areBucketNamesFetching = true;
+
+    /**
+     * Checks which type was selected on mount.
+     */
+    public async mounted(): Promise<void> {
+        this.checkedType = this.defaultType;
+        try {
+            await this.$store.dispatch(BUCKET_ACTIONS.FETCH_ALL_BUCKET_NAMES);
+            this.areBucketNamesFetching = false;
+        } catch (error) {
+            await this.$notify.error(`Unable to fetch all bucket names. ${error.message}`);
+        }
+    }
+
+    /**
+     * Closes modal.
+     */
+    public onCloseClick(): void {
+        this.$store.dispatch(ACCESS_GRANTS_ACTIONS.CLEAR_SELECTION);
+        this.onClose();
+    }
+
+    /**
+     * Toggles permissions list visibility.
+     */
+    public togglePermissions(): void {
+        if (this.showAllPermissions.show === false) {
+            this.showAllPermissions.show = true;
+            this.showAllPermissions.position = "down";
+        } else {
+            this.showAllPermissions.show = false;
+            this.showAllPermissions.position = "up";
+        }
+    }
+
+    /**
+     * Handles permissions All.
+     */
+    public toggleAllPermission(permission): void {
```

Comment:
type for permission 
and I'm sure this can be simplified. this particular method looks messy

Reply:
mb switch and more helper vars can help

Reply:
'permission' has been switched to 'type'

Spent a little bit of time trying to turn this into a switch statement, however the complexity is turning out to be the same if not more. If you have some time in the coming days we can meet and walk through what I'm finding, and brainstorm on potential solutions.